net/peer_test.go

```diff
@@ -0,0 +1,1182 @@
+	return "[]"
+}
+
+func (n *EmptyNode) RawData() []byte {
+	return nil
+}
+
+func (n *EmptyNode) Size() (uint64, error) {
+	return 0, nil
+}
+
+func (n *EmptyNode) Stat() (*ipld.NodeStat, error) {
+	return &ipld.NodeStat{}, nil
+}
+
+func createCID(doc *client.Document) (cid.Cid, error) {
+	pref := cid.V1Builder{
+		Codec:    cid.DagProtobuf,
+		MhType:   mh.SHA2_256,
+		MhLength: 0, // default length
+	}
+
+	buf, err := doc.Bytes()
+	if err != nil {
+		return cid.Cid{}, err
+	}
+
+	// And then feed it some data
+	c, err := pref.Sum(buf)
+	if err != nil {
+		return cid.Cid{}, err
+	}
+	return c, nil
+}
+
+const randomMultiaddr = "/ip4/0.0.0.0/tcp/0"
+
+func newTestNode(ctx context.Context, t *testing.T) (client.DB, *Node) {
+	store := memory.NewDatastore(ctx)
+	db, err := db.NewDB(ctx, store, db.WithUpdateEvents())
+	require.NoError(t, err)
+
+	cfg := config.DefaultConfig()
+	cfg.Net.P2PAddress = randomMultiaddr
+	cfg.Net.RPCAddress = "0.0.0.0:0"
+	cfg.Net.TCPAddress = randomMultiaddr
+
+	n, err := NewNode(
+		ctx,
+		db,
+		WithConfig(cfg),
+		WithDataPath(t.TempDir()),
+	)
+	require.NoError(t, err)
+
+	return db, n
+}
+
+func TestNewPeer_NoError(t *testing.T) {
+	ctx := context.Background()
+	store := memory.NewDatastore(ctx)
+	db, err := db.NewDB(ctx, store, db.WithUpdateEvents())
+	require.NoError(t, err)
+
+	h, err := libp2p.New()
+	require.NoError(t, err)
+
+	_, err = NewPeer(ctx, db, h, nil, nil, nil, nil, nil)
+	require.NoError(t, err)
+}
+
+func TestNewPeer_NoDB_NilDBError(t *testing.T) {
+	ctx := context.Background()
+
+	h, err := libp2p.New()
+	require.NoError(t, err)
+
+	_, err = NewPeer(ctx, nil, h, nil, nil, nil, nil, nil)
+	require.ErrorIs(t, err, ErrNilDB)
+}
+
+func TestNewPeer_WithExistingTopic_TopicAlreadyExistsError(t *testing.T) {
+	ctx := context.Background()
+	store := memory.NewDatastore(ctx)
+	db, err := db.NewDB(ctx, store, db.WithUpdateEvents())
+	require.NoError(t, err)
+
+	_, err = db.AddSchema(ctx, `type User {
+		name: String
+		age: Int
+	}`)
+	require.NoError(t, err)
+
+	col, err := db.GetCollectionByName(ctx, "User")
+	require.NoError(t, err)
+
+	doc, err := client.NewDocFromJSON([]byte(`{"name": "John", "age": 30}`))
+	require.NoError(t, err)
+
+	err = col.Create(ctx, doc)
+	require.NoError(t, err)
+
+	h, err := libp2p.New()
+	require.NoError(t, err)
+
+	ps, err := pubsub.NewGossipSub(
+		ctx,
+		h,
+		pubsub.WithPeerExchange(true),
+		pubsub.WithFloodPublish(true),
+	)
+	require.NoError(t, err)
+
+	_, err = rpc.NewTopic(ctx, ps, h.ID(), doc.Key().String(), true)
+	require.NoError(t, err)
+
+	_, err = NewPeer(ctx, db, h, nil, ps, nil, nil, nil)
+	require.ErrorContains(t, err, "topic already exists")
+}
+
+func TestStartAndClose_NoError(t *testing.T) {
+	ctx := context.Background()
+	db, n := newTestNode(ctx, t)
+
+	err := n.Start()
+	require.NoError(t, err)
+
+	db.Close(ctx)
+}
+
+func TestStart_WithKnownPeer_NoError(t *testing.T) {
+	ctx := context.Background()
+	store := memory.NewDatastore(ctx)
```

Comment:
suggest: rename `store` to `store1` to have it consistent

Reply:
✅